poutyne/framework/metrics/utils.py

```diff
@@ -16,40 +16,53 @@ def camel_to_snake(name):
     return pattern2.sub(r'\1_\2', name).lower()
 
 
-def get_metric_name(metric):
+def get_names_of_metric(metric):
     if isinstance(metric, tuple):
-        return metric
-
-    if hasattr(metric, '__name__'):
-        name = metric.__name__
+        names, metric = metric
+    elif hasattr(metric, '__name__'):
+        names = metric.__name__
     elif hasattr(metric, '__class__'):
-        name = camel_to_snake(metric.__class__.__name__)
+        names = camel_to_snake(metric.__class__.__name__)
     else:
-        name = 'unknown_metric'
+        names = 'unknown_metric'
+    return names, metric
+
 
-    name = name[:-5] if name.endswith('_loss') else name
-    return name, metric
+def flatten_metric_names(metric_names):
+    to_list = lambda names: names if isinstance(names, list) else [names]
+    return [name for names in metric_names for name in to_list(names)]
 
 
-def rename_doubles(metric_names):
-    counts = Counter(metric_names)
+def rename_doubles(batch_metrics_names, epoch_metrics_names):
+    counts = Counter(flatten_metric_names(batch_metrics_names + epoch_metrics_names))
     numbering = Counter()
```

Comment:
Why passing numbering ? it's not used outside of `rename_doubles_from_counts`.

Reply:
So that the numbering of the metrics continues for the epoch metrics. For example, see in the tests:
```python
actual = rename_doubles(['a', 'b', 'c'], ['d', 'a', 'e', 'a'])
expected = ['a1', 'b', 'c'], ['d', 'a2', 'e', 'a3']
self.assertEqual(expected, actual)
```

With numbering inside `rename_doubles_from_counts`, we would get `['a1', 'b', 'c'], ['d', 'a1', 'e', 'a2']` instead of what is expected.